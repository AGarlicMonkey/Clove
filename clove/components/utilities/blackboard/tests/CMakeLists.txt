find_package(GTest REQUIRED CONFIG)

#Blackboard
add_executable(BlackboardTest BlackboardTests.cpp)
target_link_libraries(BlackboardTest PRIVATE GTest::gtest_main CloveBlackboard)
add_test(NAME BlackboardTest COMMAND BlackboardTest)