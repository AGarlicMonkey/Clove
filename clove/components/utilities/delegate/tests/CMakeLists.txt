#SingleCastDelegate
add_executable(SingleCastDelegateTest SingleCastDelegateTests.cpp)
target_link_libraries(SingleCastDelegateTest PRIVATE CONAN_PKG::GTest CloveDelegate)
add_test(NAME SingleCastDelegateTest COMMAND SingleCastDelegateTest)

#MultiCastDelegate
add_executable(MultiCastDelegateTest MultiCastDelegateTests.cpp)
target_link_libraries(MultiCastDelegateTest PRIVATE CONAN_PKG::GTest CloveDelegate)
add_test(NAME MultiCastDelegateTest COMMAND MultiCastDelegateTest)