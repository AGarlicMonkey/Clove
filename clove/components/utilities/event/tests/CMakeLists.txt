add_executable(EventTest EventTests.cpp)
target_link_libraries(EventTest PRIVATE ${CONAN_LIBS_GTEST} CloveEvent)
add_test(NAME EventTest COMMAND EventTest)