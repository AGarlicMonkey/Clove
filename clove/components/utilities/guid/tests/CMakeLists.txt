find_package(GTest REQUIRED CONFIG)

add_executable(GuidTest GuidTests.cpp)
target_link_libraries(GuidTest PRIVATE GTest::gtest_main CloveGuid)
add_test(NAME GuidTest COMMAND GuidTest)