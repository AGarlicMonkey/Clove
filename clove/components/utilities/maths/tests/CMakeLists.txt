find_package(GTest REQUIRED CONFIG)

#Vector
add_executable(VectorTest VectorTests.cpp)
target_link_libraries(VectorTest PRIVATE GTest::gtest_main CloveMaths)
add_test(NAME VectorTest COMMAND VectorTest)

#Matrix
add_executable(MatrixTest MatrixTests.cpp)
target_link_libraries(MatrixTest PRIVATE GTest::gtest_main CloveMaths)
add_test(NAME MatrixTest COMMAND MatrixTest)