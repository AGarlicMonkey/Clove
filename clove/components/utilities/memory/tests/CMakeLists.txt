find_package(GTest REQUIRED CONFIG)

#StackAllocator
add_executable(StackAllocatorTest StackAllocatorTests.cpp)
target_link_libraries(StackAllocatorTest PRIVATE GTest::gtest_main CloveMemory)
add_test(NAME StackAllocatorTest COMMAND StackAllocatorTest)

#ListAllocator
add_executable(ListAllocatorTest ListAllocatorTests.cpp)
target_link_libraries(ListAllocatorTest PRIVATE GTest::gtest_main CloveMemory)
add_test(NAME ListAllocatorTest COMMAND ListAllocatorTest)

#PoolAllocator
add_executable(PoolAllocatorTest PoolAllocatorTests.cpp)
target_link_libraries(PoolAllocatorTest PRIVATE GTest::gtest_main CloveMemory)
add_test(NAME PoolAllocatorTest COMMAND PoolAllocatorTest)