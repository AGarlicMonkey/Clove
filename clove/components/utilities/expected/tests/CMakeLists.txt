add_executable(ExpectedTest ExpectedTests.cpp)
target_link_libraries(ExpectedTest PRIVATE CONAN_PKG::gtest CloveExpected)
add_test(NAME ExpectedTest COMMAND ExpectedTest)